### DISCUSSION TITLE
When testing for enrichment, how to keep the directionality

### DISCUSSION THREAD

**Discussion by ramirobarrantes** [zwdzwd/sesame#179] [General]
I have a set of tumor samples with and without a particular condition. For us it is of great interest to "tease apart" what genes are methylated when the condition is present versus when the condition is absent, I did as follows:

```
se <- SummarizedExperiment(betas, colData = metaData)
summary = DML(se, ~condition, BPPARAM = BiocParallel::MulticoreParam(4))
test_result = summaryExtractTest(summary)
```
and what I gather is that the effect size would reflect the directionality of interest (effect size positive when the condition is present, and negative otherwise). 

However, when I test for enrichment:

```
df <- testResult[testResult$Pval_Condition < 0.01 & abs(testResult$Est_Condition) > 0.1,]
result <- testEnrichment(df$Probe_ID, KYCG_buildGeneDBs(df$Probe_ID, max_distance=100000, platform="EPIC"),platform="EPIC")
```

and I draw a volcano plot, I only seem to be able to see one direction (see below), i..e the directionality distinction is lost. Why is that? Do I need to enrich separately for presence/absence of condition?

<img width="1055" alt="Screenshot 2024-09-12 at 5 22 29 AM" src="https://github.com/user-attachments/assets/f6db0b8c-89f8-4a54-a985-6a41cc308248">

**Comment by bethan-mallabar-rimmer** [comment]
As this question was a month ago you have probably come to a solution by now, but for anyone viewing this thread in future: the bi-directional volcano plot requires an extra argument to `testEnrichment` of `alternative = "two.sided"` . E.g. `result <- testEnrichment(df$Probe_ID, KYCG_buildGeneDBs(df$Probe_ID, max_distance=100000, platform="EPIC"), platform="EPIC", alternative="two.sided")` . Hopefully that works (I've only tested on EPICv2 data not EPIC).
Best,
Bethan
